fl4health/datasets/skin_cancer/load_data.py

```diff
@@ -51,49 +71,79 @@ def load_skin_cancer_data(
+    seed: Optional[int] = None,
+) -> Tuple[DataLoader, DataLoader, DataLoader, Dict[str, int]]:
+    """
+    Load skin cancer dataset (training, validation, and test set).
+
+    Args:
+        data_dir (Path): Directory containing the dataset files.
+        dataset_name (str): Name of the dataset to load.
+        batch_size (int): Batch size for the DataLoader.
+        split_percents (Tuple[float, float, float]): Percentages for splitting the data into train, val, and test sets.
+        sampler (Optional[LabelBasedSampler]): Sampler for the dataset. Defaults to None.
+        train_transform (Union[None, Callable]): Transformations to apply to the training data. Defaults to None.
+        val_transform (Union[None, Callable]): Transformations to apply to the validation data. Defaults to None.
+        test_transform (Union[None, Callable]): Transformations to apply to the test data. Defaults to None.
+        dataset_converter (Optional[DatasetConverter]): Converter to apply to the dataset. Defaults to None.
+        seed (Optional[int]): Random seed for shuffling data. Defaults to None.
+
+    Returns:
+        Tuple[DataLoader, DataLoader, DataLoader, Dict[str, int]]: DataLoaders for the training, validation,
+            and test sets, and a dictionary with the number of examples in each set.
+    """
+    if sum(split_percents) != 1.0:
+        raise ValueError("The split percentages must sum to 1.0")
 
     dataset_paths = {
-        "Barcelona": data_dir / "ISIC_2019" / "ISIC_19_Barcelona.json",
-        "Rosendahl": data_dir / "HAM10000" / "HAM_rosendahl.json",
-        "Vienna": data_dir / "HAM10000" / "HAM_vienna.json",
-        "UFES": data_dir / "PAD-UFES-20" / "PAD_UFES_20.json",
-        "Canada": data_dir / "Derm7pt" / "Derm7pt.json",
+        "Barcelona": data_dir.joinpath("ISIC_2019", "ISIC_19_Barcelona.json"),
+        "Rosendahl": data_dir.joinpath("HAM10000", "HAM_rosendahl.json"),
+        "Vienna": data_dir.joinpath("HAM10000", "HAM_vienna.json"),
+        "UFES": data_dir.joinpath("PAD-UFES-20", "PAD_UFES_20.json"),
+        "Canada": data_dir.joinpath("Derm7pt", "Derm7pt.json"),
     }
 
     if dataset_name not in dataset_paths:
-        log(ERROR, f"Dataset {dataset_name} not found in available datasets.")
         raise ValueError(f"Dataset {dataset_name} not found in available datasets.")
 
     dataset_path = dataset_paths[dataset_name]
 
     if not dataset_path.exists():
-        log(
-            ERROR,
+        raise FileNotFoundError(
             f"Dataset file {dataset_path} does not exist.\
-            Please follow the instructions in fl4health/datasets/skin_cancer/README.md.",
+            Please follow the instructions in fl4health/datasets/skin_cancer/README.md."
         )
-        raise FileNotFoundError(f"Dataset file {dataset_path} does not exist.")
 
-    log(INFO, f"Data directory: {str(dataset_path)}")
+    print(f"Data directory: {str(dataset_path)}")
```

Comment:
Rather than printing here, it's good to push it into the logging stream properly with `log(INFO, ...)` as you originally had 🙂